gemonlinedb/src/common/ConfigurationManager.cc

```diff
@@ -0,0 +1,167 @@
+#include "gem/onlinedb/ConfigurationManager.h"
+
+#include <cstdlib>
+
+#include <xercesc/dom/DOM.hpp>
+#include <xercesc/parsers/XercesDOMParser.hpp>
+
+#include "gem/onlinedb/ConfigurationLinker.h"
+#include "gem/onlinedb/SystemTopology.h"
+#include "gem/onlinedb/XMLConfigurationProvider.h"
+#include "gem/onlinedb/detail/FileUtils.h"
+
+namespace gem {
+    namespace onlinedb {
+
+        namespace /* anonymous */ {
+
+            /**
+             * \brief Loads a system-topology.xml file
+             */
+            DOMDocumentPtr loadXMLTopologyFile(const std::string &path)
+            {
+                using namespace detail::literals;
+                XERCES_CPP_NAMESPACE_USE
+
+                detail::XercesGuard guard; // Make sure that Xerces is loaded
+
+                // Find the schema (throws if not found)
+                auto schemaPath = detail::getFileInPath(
+                    "system-topology.xsd",
+                    "xml/schema:/opt/cmsgemos/share/gemonlinedb/xml/schema");
+
+                // Find the file (throws if not found)
+                auto filePath = detail::getFileInPath("system-topology.xml", path);
+
+                // Get an implementation
+                auto impl = DOMImplementationRegistry::getDOMImplementation("LS"_xml);
+                if (impl == nullptr) {
+                    throw std::logic_error("Cannot find a DOM implementation");
+                }
+
+                auto errorHandler =
+                    std::make_shared<detail::XercesAlwaysThrowErrorHandler>();
+
+                XercesDOMParser parser;
+                parser.setErrorHandler(errorHandler.get());
+                parser.setExternalSchemaLocation(
+                    ("https://github.com/cms-gem-daq-project/cmsgemos/tree/master/gemonlinedb/xml/schema/system-topology.xsd "
```

Comment:
This syntax defines the location of the XML schema identified by `https://github.com/cms-gem-daq-project/...`. The second argument (after the whitespace) is its path on the local filesystem. No remote access is required.

The schema URL has to be written in XML files (when they are namespace-aware like this one), the `xsd` file and the loading code. It is hence part of the data format.

In the event of a migration to Gitlab, I'm hoping that a read-only archive would be kept for historical purpose.

Reply:
Again, I would define it as a string constant in a well-known place, rather than burried somewhere in the code, so in case we need to change it, it would be more straightforward

Reply:
Then give me a `configure` step to sync it with the `xsd` and example `xml`… I can do that with any modern build system, but not the xDAQ Makefiles.

Note that changing this value would result in old files not working anymore (that's what I meant with "part of the data format").

Reply:
Jumping in late here, I wouldn't set the schema URL to the VC file, rather to a site, e.g., where we'll host, e.g., the documentation and `yum` repos (yet to be fully configured, almost certainly somewhere in https://www.cern.ch/cmsgemdaq/sw/)
Then it can be "versioned" in the event the schema itself needs to be updated.
Additionally, as it's currently just a namespace identifier, I think we can leave it as is for now (but @lmoureaux can you add a `FIXME host for schema` comment next to it so my routine searches for `FIXME` will ensure this doesn't get lost?)